cmd/mdatagen/templates/metrics.go.tmpl

```diff
@@ -263,6 +283,21 @@ func (mb *MetricsBuilder) EmitForResource(rmo ...ResourceMetricsOption) {
 	for _, op := range rmo {
 		op(rm)
 	}
+	{{ if .ResourceAttributes -}}
+	for name, val := range rm.Resource().Attributes().AsRaw() {
+		if filter, ok := mb.resourceAttributeIncludeFilter[name]; ok {
+			if !filter.Matches(val) {
+				return
+			}
+		}
+		if filter, ok := mb.resourceAttributeExcludeFilter[name]; ok {
+			if filter.Matches(val) {
+				return
+			}
+		}
```

Comment:
This behavior of combining the `include` and `exclude` looks good to me. I'm curious: what did you use as a reference? Is it aligned with what we have in the filter processor, and is it what the user would expect? This is a pretty important decision and needs to be well documented. I'm mostly talking about the priority choice. 

Also, I think this should also be extracted to the filter module. So, any other implementations get the same behavior.